pyspades/server.py

```diff
@@ -249,9 +275,15 @@ def update_network(self):
         world_update = loaders.WorldUpdate()
         # we only want to send as many items of the player list as needed, so
         # we slice it off at the highest player id
-        world_update.items = items[:highest_player_id+1]
-        self.broadcast_contained(world_update, unsequenced=True)
-
+        world_update.items = items[:max(self.players)+1]
+        if for_spectators:
+            for player in self.connections.values():
+                if (player.team is not None and player.team.spectator) or \
+                (player.world_object is None) or player.world_object.dead: # Dead players
```

Comment:
Pretty sure this first check will send frequent world updates to people downloading the map. Which is not very helpful.